src/storage/serialization/serialize_nodes.cpp

```diff
@@ -309,7 +309,7 @@ JoinCondition JoinCondition::Deserialize(Deserializer &deserializer) {
 
 void LogicalType::Serialize(Serializer &serializer) const {
 	serializer.WriteProperty<LogicalTypeId>(100, "id", id_);
-	serializer.WritePropertyWithDefault<shared_ptr<ExtraTypeInfo>>(101, "type_info", type_info_);
+	serializer.WritePropertyWithDefault<ExtraTypeInfo*>(101, "type_info", type_info_.get());
```

Comment:
Are these changes necessary?

Reply:
The change to `ExtraTypeInfo*` from `shared_ptr<ExtraTypeInfo>` yes, if I have an optional_ptr or used the referencing operator on a reference (`&to_scan` for example) it can't translate to a `shared_ptr`.

This class got caught in the crossfire because it met the conditions: `is_pointer` and `custom_serialize_property`